enterprise/management/commands/bulk_update_catalog_query_id.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+"""
+Django management command for bulk updating EnterpriseCustomerCatalog record's EnterpriseCatalogQuery's.
+"""
+
+import logging
+
+from django.core.management import BaseCommand
+
+from enterprise.models import EnterpriseCustomerCatalog, EnterpriseCatalogQuery
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Takes a new and old Enterprise Catalog Query ID, find all most recent records within the historical Enterprise
+    Catalog table where enterprise catalog query ID is the old ID provided and updates the corresponding Enterprise
+    Catalog table entries with the new ID.
+    """
+    help = "Updates specified EnterpriseCatalog's enterprise catalog query ID records with a provided new ID"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'old_id',
+            action='store',
+            help='Old catalog query ID to be replaced.'
+        )
+        parser.add_argument(
+            'new_id',
+            action='store',
+            help='New catalog query ID to replace the old one.'
+        )
+
+    def handle(self, *args, **options):
+        """
+        Entry point for management command execution.
+        """
+        LOGGER.info("Starting bulk update of EnterpriseCatalog's enterprise catalog query IDs")
+
+        old_id = options['old_id']
+        new_id = options['new_id']
+        uuids_to_change = []
+        if EnterpriseCatalogQuery.objects.filter(id=new_id).first():
+            catalogs_to_change = EnterpriseCustomerCatalog.history.filter(enterprise_catalog_query_id=old_id).order_by(
+                '-modified'
+            ).distinct()
+            for catalog in catalogs_to_change:
+                if not catalog.next_record and catalog.get_history_type_display() != 'Deleted':
+                    uuids_to_change.append(str(catalog.uuid))
```

Comment:
I don't think this actually has to be a string.
```
>>> uuids.append(catalog1.uuid)
>>> uuids.append(catalog2.uuid)
>>> EnterpriseCustomerCatalog.objects.filter(uuid__in=uuids)
<QuerySet [<EnterpriseCustomerCatalog 'All Content' with UUID '6cbd3a0e-dc96-4d67-821e-53ca39d992c0' for EnterpriseCustomer Moodle Test Enterprise>, <EnterpriseCustomerCatalog 'Test UUID' wi
```

Reply:
oh yeeeeah. Cool thx 👍